src/MetricsUploader/MetricsUploaderWindows.cpp

```diff
@@ -200,6 +201,7 @@ ErrorMessageOr<std::string> GenerateUUID() {
 }
 
 bool MetricsUploaderImpl::FillAndSendLogEvent(OrbitLogEvent partial_filled_event) const {
+  absl::MutexLock lock(&mutex_);
   if (send_log_event_addr_ == nullptr) {
```

Comment:
I suspect we still need to figure out whether we may call functions from metrics_uploader_client.dll from a different thread. I tried to google but the docs about cgo don't mention anything.

Reply:
This is a good point. It is not addressed here at all. To be on the safe side, we should probably make sure its only called from the main thread

Reply:
We could do that similar to ScopedStatus and give the ScopedMetric a pointer to an Executor. Send the metric means scheduling a task on the executor.